frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js

```diff
@@ -0,0 +1,352 @@
+  complianceCodesHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  getDropdownHSRCMComplianceCodes: CustomPropTypes.options.isRequired,
+  mineRegionOptions: CustomPropTypes.options.isRequired,
+  filterVarianceStatusOptions: CustomPropTypes.filterOptions.isRequired,
+  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
+};
+
+const joinOrRemove = (param, key) => (isEmpty(param) ? {} : { [key]: param.join(",") });
+const formatParams = ({ region = [], compliance_code = [], ...remainingParams }) => ({
+  ...joinOrRemove(region, "region"),
+  ...joinOrRemove(compliance_code, "compliance_code"),
+  ...remainingParams,
+});
+
+export class CustomHomePage extends Component {
+  constructor(props) {
+    super(props);
+    this.handleVarianceSearchDebounced = debounce(this.handleVarianceSearch, 1000);
+    this.state = {
+      params: {
+        page: String.DEFAULT_PAGE,
+        per_page: String.DEFAULT_PER_PAGE,
+        major: "",
+        region: [],
+        compliance_code: [],
+        issue_date_min: "",
+        issue_date_max: "",
+        expiry_date_max: "",
+        expiry_date_min: "",
+        search: "",
+      },
+    };
+  }
+
+  state = {
+    variancesLoaded: false,
+    params: {
+      variance_application_status_code: [],
+      page: Strings.DEFAULT_PAGE,
+      per_page: 5,
+      major: "",
+      region: [],
+      compliance_code: [],
+      issue_date_min: "",
+      issue_date_max: "",
+      expiry_date_max: "",
+      expiry_date_min: "",
+      search: "",
+    },
+  };
+
+  componentDidMount() {
+    const params = this.props.location.search;
+    if (params) {
+      this.renderDataFromURL(params);
+    } else {
+      this.props.history.push(
+        router.VARIANCE_DASHBOARD.dynamicRoute({
+          page: String.DEFAULT_PAGE,
+          per_page: String.DEFAULT_PER_PAGE,
+        })
+      );
+    }
+
+    this.props.fetchVariances(this.state.params).then(() => {
+      this.setState({ variancesLoaded: true });
+    });
+    this.props.fetchInspectors();
+    this.props.fetchMineTenureTypes();
+    this.props.fetchMineComplianceCodes();
+    this.props.fetchRegionOptions();
+    this.props.fetchMineCommodityOptions();
+    this.props.fetchVarianceStatusOptions();
+    this.props.fetchVarianceDocumentCategoryOptions();
+  }
+
+  componentWillUnmount() {
+    this.handleVarianceSearchDebounced.cancel();
+    this.setState({
+      params: {},
+    });
+  }
+
+  renderDataFromURL = (params) => {
+    const {
+      issue_date_min,
+      issue_date_max,
+      expiry_date_max,
+      expiry_date_min,
+      region,
+      compliance_code,
+      major,
+      search,
+      ...remainingParams
+    } = queryString.parse(params);
+    const format = (param) => (param ? param.split(",").filter((x) => x) : []);
+    this.setState({
+      params: {
+        issue_date_min,
+        issue_date_max,
+        expiry_date_max,
+        expiry_date_min,
```

Comment:
Any reason why these are broken-out? It looks like they can be left inside of `remainingParams`, since they're not being transformed (like `region` and `compliance_codes`)

Reply:
Good catch, fixed. :)